### PR TITLE
Add QC looking for xrefs to OMIM genes

### PR DESCRIPTION
**Pull request by matentzn**
This is a QC check in case you want it @twhetzel - I dont have a strong opinion either way. It looks for OMIM genes in Xrefs, and fails if it finds any. See #7983  for context.

### CODE REVIEW

**Review by twhetzel — APPROVED** [review]

**Comment by twhetzel** [comment]
I thought this was failing because of the existing use of gene symbols in xrefs, ie what this qc check is for, but it is failing with a warning for 11439 entries:
```
robot --catalog catalog-v001.xml merge -i reasoned.owl annotate -V [http://purl.obolibrary.org/obo/mondo/releases/`date](http://purl.obolibrary.org/obo/mondo/releases/%60date) +%Y-%m-%d`/mondo.owl -o mondo.owl
owltools  --log-error --use-catalog mondo.owl --reasoner elk --merge-equivalence-sets -P MONDO -s MONDO 100 --remove-dangling -o test_nomerge
robot --catalog catalog-v001.xml report --fail-on ERROR --profile profile.txt --print 8 -i mondo-tags-reasoned.owl -o reports/robot-report-mondo-tags-reasoned.owl.tsv
Violations: 11439
-----------------
ERROR:      0
WARN:       11439
INFO:       0

First 8 violations:
WARN	duplicate_exact_synonym	MONDO:0017304	oboInOwl:hasExactSynonym	XLOA
WARN	duplicate_exact_synonym	MONDO:0021019	oboInOwl:hasExactSynonym	XLOA
WARN	duplicate_exact_synonym	MONDO:0006015	oboInOwl:hasExactSynonym	WFS
WARN	duplicate_exact_synonym	MONDO:0018105	oboInOwl:hasExactSynonym	WFS
WARN	duplicate_exact_synonym	MONDO:0005102	oboInOwl:hasExactSynonym	UES
WARN	duplicate_exact_synonym	MONDO:0015795	oboInOwl:hasExactSynonym	UES
WARN	duplicate_exact_synonym	MONDO:0010979	oboInOwl:hasExactSynonym	TS
WARN	duplicate_exact_synonym	MONDO:00[164](https://github.com/monarch-initiative/mondo/actions/runs/10029452649/job/27717647532?pr=7984#step:4:165)55	oboInOwl:hasExactSynonym	TS
```

**Comment by matentzn** [comment]
Looking at [the check](https://github.com/monarch-initiative/mondo/actions/runs/10029452649/job/27717647532?pr=7984), it fails with these lines:

```
robot --catalog catalog-v001.xml query --use-graphs true -i tmp/omim.owl -f tsv --tdb true --query ../sparql/reports/omim-genes.sparql tmp/omim-genes.tsv
sed -i 's/[?]//g' tmp/omim-genes.tsv
sed -i 's/[<]https[:][/][/]omim[.]org[/]entry[/]/OMIM:/g' tmp/omim-genes.tsv
sed -i 's/>//g' tmp/omim-genes.tsv
tail -n +2 tmp/omim-genes.tsv > output_file && mv output_file tmp/omim-genes.tsv
grep -Ff tmp/omim-genes.tsv mondo-edit.obo | grep '^xref' > tmp/omim-gene-matches.txt || true
if [ -s tmp/omim-gene-matches.txt ]; then \
	echo "Error: xref matches found in tmp/omim-gene-matches.txt"; \
	exit 1; \
fi
Error: xref matches found in tmp/omim-gene-matches.txt
make: *** [mondo.Makefile:619: tmp/omim-gene-matches.txt] Error 1
```

Which is exactly what the check intended. `WARN` messages in the sparql check should never cause the pipeline to fail!

You approved this PR, so now i am unsure if you intended for me to fix something or just merge it - I assigned it to you so you decide if you want this test, which adds runtime of the test suite for something that may be a bit unlikely to happen in the future. So I'd rather that you keep it assigned to yourself?

File: src/ontology/mondo.Makefile [hunk 1]
```diff
@@ -600,6 +600,28 @@ mappings_fast:
...
+	sed -i 's/[<]https[:][/][/]omim[.]org[/]entry[/]/OMIM:/g' $@
+	sed -i 's/>//g' $@
+	tail -n +2 $@ > output_file && mv output_file $@
+
+# Check for occurrences of OMIM genes in MONDO,
+# Then narrow down to only xrefs
+tmp/omim-gene-matches.txt: tmp/omim-genes.tsv
+	grep -Ff $< mondo-edit.obo | grep '^xref' > $@ || true
+	if [ -s $@ ]; then \
+		echo "Error: xref matches found in $@"; \
```
Comment (matentzn) [inline src/ontology/mondo.Makefile:620]:
```suggestion
		echo "Error: xref matches found in $@"; \
		cat $@; \
```

File: src/ontology/mondo.Makefile [hunk 2]
```diff
@@ -600,6 +600,29 @@ mappings_fast:
...
```
Comment (twhetzel) [inline src/ontology/mondo.Makefile:620]:
```suggestion
		echo "FAIL: OMIM gene entry used in xref (matches found in $@)"; \
```

**Comment by twhetzel** [comment]
Too many changes in `master` now to deal with. Recreated with https://github.com/monarch-initiative/mondo/pull/8020

**Pull request closed without merging**